modules/consensus/synchronize.go

```diff
@@ -90,8 +90,14 @@ func (cs *ConsensusSet) threadedReceiveBlocks(conn modules.PeerConn) error {
 		}
 
 		// Integrate the blocks into the consensus set.
-		for _, block := range newBlocks {
-			acceptErr := cs.AcceptBlock(block)
+		for i, block := range newBlocks {
+			// Only broadcast the last block.
+			var acceptErr error
+			if !moreAvailable && i == len(newBlocks)-1 {
+				acceptErr = cs.AcceptBlock(block)
+			} else {
+				acceptErr = cs.managedAcceptBlock(block)
+			}
```

Comment:
defining a `broadcastBlock` helper function in the gateway makes sense to me, because to help with network flooding issues we're going to be sending around headers instead of blocks when broadcasts occur. Replacing the direct broadcast with the wrapped broadcast means it'll be a cleaner transition when the time comes.